frontend/components/map/legend/item/component.tsx

```diff
@@ -1,33 +1,40 @@
 import { FC } from 'react';
 
-import cx from 'classnames';
+import { X } from 'react-feather';
+
+import Button from 'components/button';
 
 import type { LegendItemProps } from './types';
 
 export const LegendItem: FC<LegendItemProps> = ({
   id,
   name,
-  description,
+  // description,
   icon,
   children,
-}: LegendItemProps) => (
-  <div key={id} className="py-2.5 px-5">
-    <div className="flex">
-      <div
-        className={cx({
-          relative: true,
-          'pl-5': icon,
-        })}
-      >
-        {icon && <div className="absolute top-0 left-0">{icon}</div>}
-        <div className="text-sm text-white font-heading">{name}</div>
+  handleCloseLegend,
+}: LegendItemProps) => {
+  return (
+    <div key={id} className="px-2 py-2.5 mb-1 bg-white rounded">
+      <div className="flex items-center justify-between">
+        <div className="">
+          <div className="flex items-center leading-[14px]">
+            {icon}
+            <span className="text-xs leading-[14px]">{name}</span>
+          </div>
+        </div>
+        <Button
+          type="button"
+          className="absolute z-50 flex items-center w-3 h-3 right-2"
+          onClick={handleCloseLegend}
+          theme="naked"
+          icon={() => <X className="text-gray-800 transition-transform hover:rotate-180" />}
```

Comment:
The icon is barely visible. I would suggest you move the icon inside the button like this:
```jsx
<Icon
  icon={X}
  className="w-4 h-4 stroke-[3px] text-gray-800 transition-transform hover:rotate-180"
/>
```

The size should be `w-4 h-4` and you can increase the stroke-width using `stroke-[3px]`.